lib/matplotlib/axes/_axes.py

```diff
@@ -3200,8 +3200,13 @@ def extract_err(err, data):
                 raise ValueError(
                     "err must be a scalar or a 1D or (2, n) array-like")
             # Using list comprehensions rather than arrays to preserve units.
-            low = [v - e for v, e in cbook.safezip(data, a)]
-            high = [v + e for v, e in cbook.safezip(data, b)]
+            for e in [a, b]:
+                if len(data) != len(e):
+                    raise ValueError(
+                        f"The lengths of the data ({len(data)}) and the "
+                        f"error {len(e)} do not match")
```

Comment:
Note that these can be either x or y errors (they both go through this helper function) so we can't include the actual parameter names here.